src/lib/integration_test_local_engine/mina_automation.ml

```diff
@@ -0,0 +1,462 @@
+open Core
+open Async
+open Currency
+open Signature_lib
+open Mina_base
+open Integration_test_lib
+
+let version = "3.9"
+
+module Network_config = struct
+  module Cli_inputs = Cli_inputs
+
+  type docker_volume_configs = {name: string; data: string}
+  [@@deriving to_yojson]
+
+  type block_producer_config =
+    { name: string
+    ; id: string
+    ; keypair: Network_keypair.t
+    ; public_key: string
+    ; private_key: string
+    ; keypair_secret: string
+    ; libp2p_secret: string }
+  [@@deriving to_yojson]
+
+  type snark_coordinator_configs =
+    {name: string; id: string; public_key: string; snark_worker_fee: string}
+  [@@deriving to_yojson]
+
+  type docker_config =
+    { version: string
+    ; stack_name: string
+    ; coda_image: string
+    ; docker_volume_configs: docker_volume_configs list
+    ; block_producer_configs: block_producer_config list
+    ; snark_coordinator_configs: snark_coordinator_configs list
+    ; log_precomputed_blocks: bool
+    ; archive_node_count: int
+    ; mina_archive_schema: string
+    ; runtime_config: Yojson.Safe.t
+          [@to_yojson fun j -> `String (Yojson.Safe.to_string j)] }
+  [@@deriving to_yojson]
+
+  type t =
+    { mina_automation_location: string
+    ; keypairs: Network_keypair.t list
+    ; debug_arg: bool
+    ; constants: Test_config.constants
+    ; docker: docker_config }
+  [@@deriving to_yojson]
+
+  let expand ~logger ~test_name ~(cli_inputs : Cli_inputs.t) ~(debug : bool)
```

Comment:
We can work later to improve the code-sharing between this and the existing cloud implementation (there are some shared steps).